### ISSUE TITLE
Linux - Crash when adding samples folder

### ISSUE THREAD

**Issue by crshrprt** [jonasblome/Saempl#3]
Linux - Debian Sid

When I try to add a samples folder to scan the plugins crashes.

I tried running it with `carla-single vst3 ~/.vst3/Saempl.vst3` to get at least some message displayed on the terminal when the plugin crashes.

The error is:
`Floating point exception`

**Comment by jonasblome** [comment]
hm, I'll take a look, maybe there's a division by zero or something. can you get any more information on that? did you try running the app in standalone mode? maybe that provides more info on the error.

**Comment by jonasblome** [comment]
try it again now, i tried to avoid any possible divisions by zero.

**Comment by crshrprt** [comment]
Now it starts Synching the sample libray in both VST3 and standalone.

VST3 version however crashes as soon the scan is finished with:
```
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  15 (X_QueryTree)
  Resource id in failed request:  0x820606
  Serial number of failed request:  5972
  Current serial number in output stream:  5972
[carla] Carla assertion failure: "engine == nullptr" in file ../backend/CarlaHostImpl.hpp, line 87
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
JUCE Assertion failure in juce_Singleton.h:50
```
Standalone seems to load samples fine most of the times, but crashed a couple of times with "segmentation error".

**Comment by jonasblome** [comment]
okay, ill try to find the issue with the singleton thing. can you give me any other info on the segmentation error thing?

**Comment by jonasblome** [comment]
i have some issues myself with building and running the au and vst schemes. next week ill copy the files to a new project and see if that fixes anything. thanks for the help you gave already!

**Comment by jonasblome** [comment]
hmm, does it work with an mp3 file for example?

**Comment by crshrprt** [comment]
This is what I get with Saempl standalone (Debug build) when I select a folder with a single mp3:
```
JUCE v8.0.12
JUCE Assertion failure in SampleLibrary.cpp:167
JUCE Assertion failure in SampleLibrary.cpp:167
JUCE Assertion failure in SampleLibrary.cpp:167
JUCE Assertion failure in SampleLibrary.cpp:167
JUCE Assertion failure in juce_AudioFormatReaderSource.cpp:44
Floating point exception   
./Saempl
```

**Comment by jonasblome** [comment]
alright, could you try again?

**Comment by crshrprt** [comment]
No more JUCE assertion failure warnings but it still crashes:

`
JUCE v8.0.12
Floating point exception   ./Saempl
`

**Comment by crshrprt** [comment]
Yes, I scanned a few directories with a few hundreds .wav files, it seems to work now. 
However, a couple of times, it crashed with a segmentation error during or at the end of the scan. 
At least it doesn't crash as soon the scanning begins anymore.
